### PR TITLE
Small blockchain processor optimizations

### PR DESCRIPTION
**Pull request by JustinTArthur**
These are some trivial adjustments to data types, iterations, and logic in the blockchain processor and storage engine.

The changes result in about a 1.17x performance bump in blocks per second for me on Python 2.7.11 on OS X.

### CODE REVIEW

File: src/blockchain_processor.py [hunk 1]
```diff
@@ -489,15 +480,13 @@ def do_unsubscribe(self, method, params, session):
                 if session in l:
                     print_log("error rc!!")
                     self.shared.stop()
-                if l == []:
-                    self.watched_addresses.pop(addr)
```
Comment (JustinTArthur) [inline src/blockchain_processor.py]: @ecdsa let me know if this makes sense. As far as I could tell, this code is unreachable due to the `if not l: return` statement before it.
 `

Comment (JustinTArthur) [inline src/blockchain_processor.py]: n/m, didn't realize what the session removal logic was doing above it. Reverting this change.

**Comment by JustinTArthur** [comment]
I've tested these changes from a fresh db as well as a full one. Connected with a 2.5.4 client and exacted some transactions.

**Comment by shsmith** [comment]
Thanks for your effort. I have been running your patch without issue.

**Pull request merged**